allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -128,11 +128,12 @@ def forward(self,  # type: ignore
             target_mask = None
 
         initial_score = Variable(embedded_input.data.new(1).fill_(0))
+        initial_state = (final_encoder_output, decoder_context, None)
```

Comment:
I think it would be cleaner to compute the initial `attended_question` here itself instead of having the conditional in `WikitablesDecoderStep.take_step`.

Reply:
Yes, that would be nice, but it breaks the abstractions we have here - you would need to know the parameters of the `DecoderStep` in order to do that in the parser's `forward` method directly.  I think having the clean separation of parameters and logic between the parser and the decoding step is worth having the slight ugliness on the first timestep.

Reply:
Well...  I suppose we could have a method that's `compute_question_attention`, or something, given a hidden state, and have `forward` call that method for the first iteration.  That'd probably work, and be cleaner.  I'll do that.